obspy/io/sac/sactrace.py

```diff
@@ -1004,22 +1004,20 @@ def reftime(self, new_reftime):
         try:
             old_reftime = self.reftime
 
-            # find the milliseconds and leftover microseconds for new reftime
-            _, rem_microsecs = _ut.split_microseconds(new_reftime.microsecond)
-
             # snap the new reftime to the most recent milliseconds
             # (subtract the leftover microseconds)
-            new_reftime.microsecond -= rem_microsecs
+            ns = new_reftime.ns
+            time = UTCDateTime(ns=(ns - ns % 1000000))
```

Comment:
time is a python module - you should avoid it as parameter/variable name if possible